minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/MissionHandlers/FileBasedPerformanceProducerImplementation.java

```diff
@@ -54,13 +58,42 @@ public boolean parseParameters(Object params)
     }
 
     @Override
-    public void step(double reward, boolean done) {
+    public void step(double reward, boolean done, JsonObject info) {
         if(PerformanceHelper.performanceEnabled()){
             if(currentEpisodeJson != null && statusJson != null){
                 currentEpisodeJson.getAsJsonArray("rewards").add(new JsonPrimitive(reward));
                 currentEpisodeJson.addProperty("numTicks", 
                     currentEpisodeJson.get("numTicks").getAsLong() + 1L);
-                
+
+                // Update the damage type
+                if (info != null && info.has("damage_source")) {
+                    // Record damage type
+                    if (info.get("damage_source").getAsJsonObject().has("damage_type")){
```

Comment:
suggestion: Refactor to have local variable `JsonObject dmgSource = info.get("damage_source").getAsJsonObject()` to reuse between if conditionals